### PR TITLE
Add base test suite

### PR DESCRIPTION
**Pull request by dwelch0**
**Summary**: Begin transition to a real application by defining some unit tests.

Included changes:
* tests to validate processing of input yaml/json config file
* tests to validate vault client initialization and retrieval of appsre TF credentials

Note: addressing testing of terraform utility will be separate effort due to complexity.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -25,7 +25,7 @@ clean:
 ############
 
 .PHONY: build
-build:
+build: vet
```
Comment (cubismod) [inline Makefile:28]: adding a note for my own sanity and for future reviewers who get confused with Makefiles, the `vet` step here is a prerequisite for running `build`: https://www.gnu.org/software/make/manual/html_node/Rule-Syntax.html

File: pkg/vault_test.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestInitVaultClient(t *testing.T) {
+	mockedToken := "65b74ffd-842c-fd43-1386-f7d7006e520a"
+	vaultMock := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```
Comment (cubismod) [inline pkg/vault_test.go:16]: TIL about this https://pkg.go.dev/net/http/httptest@go1.20.5, very cool!

**Comment by cubismod** [comment]
Shouldn't you update https://github.com/app-sre/terraform-repo-executor/blob/main/pr_check.sh to execute these newly defined tests in PR check?

**Comment by dwelch0** [comment]
@cubismod [this change](https://github.com/app-sre/terraform-repo-executor/pull/10/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R28) will include `vet` (and indirectly the `test`) when `build` is executed. Validated by proceeding output existing in latest build:
```
17:12:04 #11 12.48 === RUN   TestProcessConfig
17:12:04 #11 12.48 === RUN   TestProcessConfig/valid_yaml_returns_no_error_and_actual_equals_expected
17:12:04 #11 12.48 === RUN   TestProcessConfig/valid_json_returns_no_error_and_actual_matches_expected
17:12:04 #11 12.48 === RUN   TestProcessConfig/invalid_yaml_returns_error
17:12:04 #11 12.48 --- PASS: TestProcessConfig (0.00s)
17:12:04 #11 12.48     --- PASS: TestProcessConfig/valid_yaml_returns_no_error_and_actual_equals_expected (0.00s)
17:12:04 #11 12.48     --- PASS: TestProcessConfig/valid_json_returns_no_error_and_actual_matches_expected (0.00s)
17:12:04 #11 12.48     --- PASS: TestProcessConfig/invalid_yaml_returns_error (0.00s)
17:12:04 #11 12.48 === RUN   TestInitVaultClient
17:12:04 #11 12.48 --- PASS: TestInitVaultClient (0.00s)
17:12:04 #11 12.48 === RUN   TestGetVaultTfSecret
17:12:04 #11 12.48 --- PASS: TestGetVaultTfSecret (0.00s)
17:12:04 #11 12.48 PASS
```

**Comment by cubismod** [comment]
> @cubismod [this change](https://github.com/app-sre/terraform-repo-executor/pull/10/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R28) will include `vet` (and indirectly the `test`) when `build` is executed. Validated by proceeding output existing in latest build:
> 
> ```
> 17:12:04 #11 12.48 === RUN   TestProcessConfig
> 17:12:04 #11 12.48 === RUN   TestProcessConfig/valid_yaml_returns_no_error_and_actual_equals_expected
> 17:12:04 #11 12.48 === RUN   TestProcessConfig/valid_json_returns_no_error_and_actual_matches_expected
> 17:12:04 #11 12.48 === RUN   TestProcessConfig/invalid_yaml_returns_error
> 17:12:04 #11 12.48 --- PASS: TestProcessConfig (0.00s)
> 17:12:04 #11 12.48     --- PASS: TestProcessConfig/valid_yaml_returns_no_error_and_actual_equals_expected (0.00s)
> 17:12:04 #11 12.48     --- PASS: TestProcessConfig/valid_json_returns_no_error_and_actual_matches_expected (0.00s)
> 17:12:04 #11 12.48     --- PASS: TestProcessConfig/invalid_yaml_returns_error (0.00s)
> 17:12:04 #11 12.48 === RUN   TestInitVaultClient
> 17:12:04 #11 12.48 --- PASS: TestInitVaultClient (0.00s)
> 17:12:04 #11 12.48 === RUN   TestGetVaultTfSecret
> 17:12:04 #11 12.48 --- PASS: TestGetVaultTfSecret (0.00s)
> 17:12:04 #11 12.48 PASS
> ```

Gotcha I see what you're saying. I took another look at the PR check output and see the test output before build in there. LGTM

**Review by cubismod — APPROVED** [review]

**Pull request merged**